tests/src/test_cases/widgets/test_scale.c

```diff
@@ -222,6 +222,64 @@ void test_scale_render_example_4(void)
     TEST_ASSERT_EQUAL_SCREENSHOT("scale_4.png");
 }
 
+static void draw_event_cb(lv_event_t * e)
+{
+    lv_obj_t * obj = lv_event_get_target(e);
+    lv_draw_task_t * draw_task = lv_event_get_draw_task(e);
+    lv_draw_dsc_base_t * base_dsc = lv_draw_task_get_draw_dsc(draw_task);
+    lv_draw_label_dsc_t * label_draw_dsc = lv_draw_task_get_label_dsc(draw_task);
+    if(base_dsc->part == LV_PART_INDICATOR) {
+        if(label_draw_dsc) {
+            const lv_color_t color_idx[7] = {
+                lv_palette_main(LV_PALETTE_RED),
+                lv_palette_main(LV_PALETTE_ORANGE),
+                lv_palette_main(LV_PALETTE_YELLOW),
+                lv_palette_main(LV_PALETTE_GREEN),
+                lv_palette_main(LV_PALETTE_CYAN),
+                lv_palette_main(LV_PALETTE_BLUE),
+                lv_palette_main(LV_PALETTE_PURPLE),
+            };
+            uint8_t major_tick = lv_scale_get_major_tick_every(obj);
+            label_draw_dsc->color = color_idx[base_dsc->id1 / major_tick];
+            lv_snprintf((char *)label_draw_dsc->text,
+                        sizeof(label_draw_dsc->text),
```

Comment:
Here, the CI error is for the same reason, using lv_strlen(label_draw_dsc->text) + 1 is fine. Using +3 would also trigger an error, while +1 works correctly.

Reply:
Maybe it is because we allocate X amount of memory in the regular scale procedures and then we reuse the memory for a bigger buffer, so memory gets corrupted.

Reply:
Resolved with @kisvegabor fix